wallets/index.js

```diff
@@ -221,7 +225,38 @@ export function useWallet (name) {
 export function useSendWallets () {
   const { wallets } = useWallets()
   // return all enabled wallets that are available and can send
-  return wallets
+  return useMemo(() => wallets
     .filter(w => !w.def.isAvailable || w.def.isAvailable())
-    .filter(w => w.config?.enabled && canSend(w))
+    .filter(w => w.config?.enabled && canSend(w)), [wallets])
+}
+
+function RetryHandler ({ children }) {
+  const failedInvoices = useFailedInvoices()
+  const waitForWalletPayment = useWalletPayment()
+  const invoiceHelper = useInvoice()
+
+  useEffect(() => {
+    (async () => {
+      for (const invoice of failedInvoices) {
+        // TODO: don't retry forever
+        const newInvoice = await invoiceHelper.retry(invoice)
+        waitForWalletPayment(newInvoice).catch(console.error)
+      }
+    })()
+  }, [failedInvoices, invoiceHelper, waitForWalletPayment])
+
+  return children
+}
+
+function useFailedInvoices () {
+  const wallets = useSendWallets()
+
+  // TODO: use longer poll interval in prod?
```

Comment:
Polling for failed invoices every second was nice during development, but I think in prod every 30s is enough, so we'd want to use `NORMAL_POLL_INTERVAL`.